R/inference_api.R

```diff
@@ -0,0 +1,745 @@
+#' }
+#'
+#' @examples
+#' \dontrun{
+#' library(tidybayes)
+#' library(tidyverse)
+#'
+#' login()
+#'
+#' # ---- Re-compute SLD_hat using overall parameter estimates ----
+#' d <- fetch_biomarker_params(run_id, level = 'overall')
+#'
+#' # compute log-sld-hat overall from parameters:
+#' stein_fojo <- function(bas_sld, f, kg, ks, t) {
+#'   sld_hat <- dplyr::case_when(t < 0 ~ bas_sld * exp(kg * t),
+#'                               t == 0 ~ bas_sld,
+#'                               t > 0 ~ bas_sld * ((1-f) * exp(kg * t) + f * exp(-1*ks * t)))
+#'                               }
+#'
+#' d %>%
+#'   spread(.variable, .value) %>%
+#'   select(f, kg, ks) %>%
+#'   expand_grid(t = seq(from = -10, to = 500, by = 10)) %>%
+#'   dplyr::mutate(sld_hat = stein_fojo(f = f, kg = kg, ks = ks, t = t, bas_sld = 1))
+#'
+#' # ---- summarize sampling quality for trial-arm-level parameters ----
+#'
+#' library(posterior)
+#' d <- fetch_biomarker_params(run_id, level = 'trial_arm', return = 'draws')
+#'
+#' d %>% spread(.variable, .value) %>%
+#'    group_by(.level, trial_arm_id, .type, run_id) %>%
+#'    group_modify(~ summarise_draws(.x))
+#' }
+#'
+#' @param run_id (str) [required] One or several model run_ids. See \code{\link{find_runs}} for a list of runs available.
+#' @param level (str) The level at which to return predicted values. One of: subject, trial_arm, or overall. Default value is per subject.
+#' @param return (str) The type of summary to return. One of: median, quantiles, intervals, or draws. Default: median
+#' @param type (str) Whether to return posterior or prior predictions. Default: posterior
+#' @param project (str) The name of the project to which the run_id belongs.
+#' @param project_version_id (str) The specific project_version_id to which the run_id belongs. Defaults to the most recent project_version_id if none provided.
+#'
+#' @return data.frame in tidy format, with one record per parameter, run_id, and summarized level. See notes for specific details about each return type.
+#' @export
+fetch_biomarker_params <- function(run_id,
+                                   level = c('subject', 'trial_arm', 'overall'),
+                                   return = c('median', 'quantiles', 'intervals', 'draws'),
+                                   type = c('posterior', 'prior'),
+                                   project = NULL, project_version_id = NULL) {
+  pv_id <- .process_project_inputs(project = project, project_version_id = project_version_id)
+  level <- match.arg(level)
+  return <- match.arg(return)
+  type <- match.arg(type)
+
+  parlist <- .get_pars_by_type(type = 'biomarker_params', level = level, project_version_id = pv_id)
+  .fetch_pars_by_type(parlist = parlist, return = return, run_id = run_id, type = type, project_version_id = pv_id, level = level)
+}
+
+#' Fetch inferences for association states
```

Comment:
I don't think this description works... everything else was also fetching "inferences" -- not sure why this one is different. Maybe something that would work better is:

`Fetch the association states from the biomarker model` or `Fetch the association states linking biomarker model and the survival model`